### ISSUE TITLE
Skip feature selection for LSTM in HPO tuner

### ISSUE THREAD

**Issue by Acusick1** [Acusick1/odds#103]
**Goal**: Add guard in `create_objective()` to skip `top_k_features` tuning when strategy is LSTM, preventing runtime errors from dimension mismatch.

**Complexity**: 🟢 Low

**Location**:
- `packages/odds-analytics/odds_analytics/training/tuner.py`

**Background**:
The feature selection system operates on 2D data `(n_samples, n_features)`, but LSTM uses 3D sequences `(n_samples, timesteps, n_features)`. The current `create_objective()` function has feature subsetting logic (lines 884-900) that would fail or produce incorrect results for LSTM:

```python
# This assumes 2D arrays - breaks for LSTM 3D sequences
trial_X_train = trial_X_train[:, selected_indices]  # Line 889
```

**Design Decision**:
Feature selection is **intentionally excluded** for LSTM because:
1. **Dimension mismatch**: Selectors (XGBoost, RF, ExtraTrees) expect 2D input
2. **LSTM handles this natively**: Gating mechanisms learn to ignore irrelevant features
3. **Better NN alternatives exist**: Gradient-based saliency, attention mechanisms
4. **Regularization is more effective**: Dropout/weight decay for small datasets

If LSTM feature importance is needed in the future, it should be implemented separately using NN-appropriate methods (attention weights, integrated gradients, etc.).

**Scope**:

1. **Add strategy check** before feature selection in `create_objective()`:
   ```python
   # Around line 849
   is_lstm = strategy_type in ("lstm", "lstm_line_movement")
   
   if tuning_top_k_features and config.training.feature_selection.enabled and not is_lstm:
       # existing feature selection logic...
   elif tuning_top_k_features and is_lstm:
       logger.warning(
           "feature_selection_skipped_for_lstm",
           message="top_k_features tuning is not supported for LSTM strategies. "
           "LSTM uses gating mechanisms for implicit feature weighting.",
       )
   ```

2. **Add config validation** in `MLTrainingConfig` or `TuningConfig`:
   - Warn if `top_k_features` is in search space AND strategy is LSTM
   - Could be a `model_validator` in Pydantic

3. **Update docstring** for `create_objective()`:
   - Document that feature selection is skipped for LSTM
   - Explain the reasoning briefly

**Success criteria**:
- [ ] `create_objective()` skips feature selection for LSTM strategies
- [ ] Warning logged when LSTM + `top_k_features` combination detected
- [ ] No runtime errors when running LSTM HPO with feature selection config
- [ ] Docstring documents the behavior

**Testing**:
- Unit test: LSTM strategy with `top_k_features` in search space should log warning and proceed without feature subsetting
- Integration test: Full LSTM HPO run should complete successfully

**Prerequisites**:
- None

**Related**:
- #90 - LSTM CV/early stopping (references this decision)
- #100 - Original feature selection implementation

**Related files**:
- `packages/odds-analytics/odds_analytics/training/tuner.py` - Main file to update
- `packages/odds-analytics/odds_analytics/training/config.py` - Optional validation
